packages/parameters/src/ssm/SSMProvider.ts

```diff
@@ -790,6 +832,25 @@ class SSMProvider extends BaseProvider {
     return undefined;
   }
 
+  protected async _set(
```

Comment:
I think we can move the implementation in the main `set()` method. At least for now we don't have other interactions with other classes so having it here doesn't necessarily add much.

If we iterate on the inheritance/hierarchy of classes in the future, we can still move it out then.